### PR TITLE
feat(cms): converting the site to be a cms vs static site

### PR DESCRIPTION
**Pull request by jsundquist**
Take a look over the files and let me know before we merge it in. Let me know if there is anything that you think needs to be changed. Note that you need to create an `.env` file locally with the items within the `.env-template`. The same environment variables would need to be created on AWS as well.

### CODE REVIEW

File: models/Sponsor.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+Sponsor.add({
+	company: {type: String, required: true},
+	level: { type: Types.Relationship, ref: 'SponsorLevel' },
+	logo: {type: Types.File, storage: storage, thumb: true},
+	description: {type: Types.Html, wysiwyg: true},
+	website: {type: String},
+	facebook: {type: String},
+	github: {type: String},
+	googleplus: {type: String},
```
Comment (mwillbanks) [inline models/Sponsor.js]: G+ is dead :)

File: .env-template [hunk 2]
```diff
@@ -0,0 +1,3 @@
+COOKIE_SECRET={{cookieSecret}}
+MONGO_URI={{mongoUri}}
+
```
Comment (mwillbanks) [inline .env-template:3]: Add:
`CLOUDINARY_URL={{cloudinaryUrl}}`

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# This file is for unifying the coding style for different editors and IDEs
+# editorconfig.org
+root = true
+
+[*]
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = false
+insert_final_newline = true
+indent_style = tab
```
Comment (mwillbanks) [inline .editorconfig]: indent_style conflicts with the airbrb styleguide which you have for eslint, please change to spaces.
https://github.com/airbnb/javascript#whitespace

File: keystone.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+// Require keystone
+var keystone = require('keystone');
+var handlebars = require('express-handlebars');
+
+// Initialise Keystone with your project's configuration.
+// See http://keystonejs.com/guide/config for available options
+// and documentation.
+
+keystone.init({
+	'name': 'Midwest PHP',
```
Comment (mwillbanks) [inline keystone.js]: Midwest PHP 2019

File: keystone.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+	'brand': 'Midwest PHP',
+
+	// 'sass': 'public',
```
Comment (mwillbanks) [inline keystone.js]: Do you plan to keep the sass or not?  If so then the commented out stuff is fine but otherwise remove.

File: keystone.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+	'auto update': true,
+	'session': true,
+	'auth': true,
+	'user model': 'User',
+});
+
+// Load your project's Models
+keystone.import('models');
+
+keystone.mongoose.set('debug', true);
```
Comment (mwillbanks) [inline keystone.js]: Set this based on the environmental .env configuration.

**Review by mwillbanks — CHANGES_REQUESTED** [review]
A few larger items:
1. Remove console.log statements.
2. Provide a README.md with how to get started.
3. The editor config and styleguide do not match, we should fix that before we proceed.
4. Use const/let vs. var.

**Comment by mwillbanks** [comment]
FYI, I'll issue a PR to this branch with a lot of the eslint fixes.

File: models/Sponsor.js (hunk 1 above)
Comment (jsundquist) [reply models/Sponsor.js]: True, this was mainly coming from last years sponsors and the social media locations they provided us.

File: .env-template (hunk 2 above)
Comment (jsundquist) [reply .env-template:3]: Right now the site is setup to use local file storage vs uploading the file to cloudinary. Do we want to setup a cloudinary account to upload images to?

File: keystone.js (hunk 5 above)
Comment (jsundquist) [reply keystone.js]: Nope, will yank that.

File: .env-template (hunk 2 above)
Comment (mwillbanks) [reply .env-template:3]: I don't actually, it was required when I setup the instance locally, we can completely disable it by removing the mention of them from the models.

Comment (mwillbanks) [reply .env-template:3]: Example:
```
  heroImage: { type: Types.CloudinaryImage },
  images: { type: Types.CloudinaryImages },
``

Will cause keystone to throw an error if the URL is not defined.

Comment (jsundquist) [reply .env-template:3]: Removed the gallery model and pages as we are not using this functionality now or possibly in the future.

File: .editorconfig (hunk 3 above)
Comment (jsundquist) [reply .editorconfig]: Updated to be indent_style=space to match the airbnb styles

File: keystone.js (hunk 4 above)
Comment (jsundquist) [reply keystone.js]: Updated to be Midwest PHP 2019

File: keystone.js (hunk 5 above)
Comment (jsundquist) [reply keystone.js]: Removed the commented out code

File: keystone.js (hunk 6 above)
Comment (jsundquist) [reply keystone.js]: Removed the code as it was only being used when I was testing locally previously.

**Comment by jsundquist** [comment]
> Remove console.log statements.

Only console log item was within the update script. Leaving this one in.

> Provide a README.md with how to get started.

Added in the README.md

> The editor config and styleguide do not match, we should fix that before we proceed.

Updated

> Use const/let vs. var.

Updated in your pull request

**Review by mwillbanks — APPROVED** [review]

**Pull request merged**